### ISSUE TITLE
no-console

### ISSUE THREAD

**Issue by SimonLab** [dwyl/goodparts#16]
console.log are usefull and can give indication to the dev or users, so we want to keep it

**Comment by eliascodes** [comment]
Fair enough if this is a eslint config for development. May be worth thinking about having a separate config file for production builds with console logs forbidden.

**Comment by nelsonic** [comment]
Gents I have to _disagree_ here that we should _allow_ `console.log` In my experience unless we are debugging something _specific_ we should _remove_ (_or at least comment out all console.log_) if we want to prevent a _lot_ of noise in the application logs. 
I would _prefer_ this rule to be `on` and then if people _**really need**_ console.log in their code to debug something in Production they would state that explicitly in their commit message e.g:
```js
git commit -m '[WiP] :bug: attempting to fix issue #791 - database hangs on transaction approval' -n
```

**Comment by eliascodes** [comment]
@nelsonic what about uses of `console.log` that are not for debugging. For example, logging information on the server side. Should this then always be done with specialist logging module (e.g. [winston](https://github.com/winstonjs/winston))? Or do you think even that should be avoided?

**Comment by nelsonic** [comment]
@eliascodes yes, logging on the server should be done using a logging module. We should do a mini-investigation into this. I've used [Winston](https://github.com/winstonjs/winston) & [Bunyan](https://github.com/trentm/node-bunyan) but I think we should try using https://github.com/hapijs/good and (HTTP) streaming logs to an ElasticSearch instance where they can be viewed in Kibana ... or, do have you come across a simpler solution? 💭

**Comment by SimonLab** [comment]
Fair enough, I've updated the rule value to never allow console.log

**Comment by nelsonic** [comment]
We have a simple (but decent) example of logging using `good`: https://github.com/dwyl/learn-hapi#logging-with-good
